pyracf/common/security_result.py

```diff
@@ -9,10 +10,14 @@
 class SecurityResult:
     """Generic Security Result Parser."""
 
-    def __init__(self, result_xml: str) -> None:
+    def __init__(
+        self, result_xml: str, running_userid: Union[str, None] = None
+    ) -> None:
         self.__result = XMLParser.fromstring(result_xml)
         self.__result_dictionary = {"securityResult": {}}
         self.__extract_results()
+        if running_userid is not None:
+            self.__result_dictionary["securityResult"]["runningUserid"] = running_userid
```

Comment:
Does it make sense to have it default to the current user for audit purposes? You can use `getpass` to do this in Python.

https://stackoverflow.com/questions/842059/is-there-a-portable-way-to-get-the-current-username-in-python

Reply:
Also, I don't know if it makes sense to change `runningUserid` to something like `authorityUsed` so it is clear that it is saying that the following userid is the one who's authority was used to make the request?

Reply:
It could cause problems with RACF permissions if we were to force pyRACF into utilizing running_userid when it would not be intended. We could get around this with more checks on the userid, but that feels unnecessary.Our "audit purposes" would really only add the running userid to debug logging or structure output which I don't know really helps. 

I could consider a name change to authorityUsed, but I like the idea of keeping it similar to IRRSMO00's RunAsUserID so that we can reference their documentation for the authority needed and have that make sense. We can discuss in more detail.

Reply:
I am okay with keeping `runningUserid`. However, my suggestion is not to always run in run as userid mode. my suggestion is that if `running_userid` is not set, then it is assumed that the current user running the program is the "running userid". Then we can just use get pass to get that userid and add it to the result dictionary:

```python
        if running_userid is None:
            self.__result_dictionary["securityResult"]["runningUserid"] = getpass.getuser()
        else:
            self.__result_dictionary["securityResult"]["runningUserid"] = running_userid
```

Also it is probably a good idea to standardize to lowercase for the values in the security result dictionary:

```python
self.__result_dictionary["securityResult"]["runningUserid"] = running_userid.lower()
```

Reply:
That would work functionally but based on the current call to RACF, this would drive IRRSMO00 to try and pass that userid as RunAsUserID which I could see as an issue. We could set up a check against the currently running user, or some sort of other flag involved in "clearing" or "setting" this value, but I am on the fence about whether that's necessary for this. I did cast it to lowercase though.

Reply:
Oh, I just realized where you were proposing the codechange. I get it. I can look at that change.

Reply:
Testing issues resolved using a mock.